cmd/sign.go

```diff
@@ -78,20 +76,21 @@ var signCmd = &cobra.Command{
 		}
 		fmt.Println("\nReceived OpenID Scope retrieved for account:", email)
 
-		key, pub, err := keymgmt.GeneratePrivateKey("ecdsaP256")
+		signer, err := signature.NewDefaultECDSASignerVerifier()
+		if err != nil {
+			return err
+		}
+		pub, err := signer.PublicKey(ctx)
 		if err != nil {
 			return err
 		}
 
-		privKey := key.(*ecdsa.PrivateKey)
-		pubKey := pub.(*ecdsa.PublicKey)
-		pubBytes, err := x509.MarshalPKIXPublicKey(pubKey)
+		pubBytes, err := x509.MarshalPKIXPublicKey(pub)
 		if err != nil {
 			return err
 		}
 
-		emailHash := sha256.Sum256([]byte(email))
-		proof, err := ecdsa.SignASN1(rand.Reader, privKey, emailHash[:])
+		proof, _, err := signer.Sign(ctx, []byte(email))
```

Comment:
Does signer.Sign also generate a hash as well as sign?

Reply:
Yeah, my understanding is that it's "signature, signed, error", some algorithms (ed25519) do hashing on their own, others require you to prehash, some like fido2 actually mutate the input payload in other ways before signing.

Reply:
ok, got it now.

Reply:
Yup, what Dan said.